model-optimizer/mo/main.py

```diff
@@ -98,35 +98,44 @@ def print_argv(argv: argparse.Namespace, is_caffe: bool, is_tf: bool, is_mxnet:
 
 def get_moc_frontends(argv: argparse.Namespace):
     fem = argv.feManager
-    available_moc_front_ends = []
-    moc_front_end = None
-
-    # TODO: in future, check of 'use_legacy_frontend' in argv can be added here (issue 61973)
-    force_use_legacy_frontend = False
-
-    if fem and not force_use_legacy_frontend:
-        available_moc_front_ends = fem.get_available_front_ends()
-        if argv.input_model:
-            if not argv.framework:
-                moc_front_end = fem.load_by_model(argv.input_model)
-                # skip onnx frontend as not fully supported yet (63050)
-                if moc_front_end and moc_front_end.get_name() == "onnx":
-                    moc_front_end = None
-                if moc_front_end:
-                    argv.framework = moc_front_end.get_name()
-            elif argv.framework in available_moc_front_ends:
-                moc_front_end = fem.load_by_framework(argv.framework)
+
+    # Read user flags:
+    use_legacy_frontend = argv.use_legacy_frontend
+    use_new_frontend = argv.use_new_frontend
+
+    if not fem or use_legacy_frontend:
+        return None, []
+
+    available_moc_front_ends = fem.get_available_front_ends()
+
+    if not argv.framework and argv.input_model:
+        moc_front_end = fem.load_by_model(argv.input_model)
+        if not moc_front_end:
+            return None, available_moc_front_ends
+        argv.framework = moc_front_end.get_name()
+    elif argv.framework in available_moc_front_ends:
+        moc_front_end = fem.load_by_framework(argv.framework)
+    else:
+        return None, []
+
+    # Set which frontend to use by default, values should be 'new' or 'legacy'
+    frontend_defaults = {
+        'onnx': 'legacy',
+    }
+    # Disable MOC frontend if default is set to legacy and no user override
+    if frontend_defaults.get(moc_front_end.get_name()) == 'legacy' and not use_new_frontend:
+        moc_front_end = None
 
     return moc_front_end, available_moc_front_ends
 
 
 def arguments_post_parsing(argv: argparse.Namespace):
     moc_front_end, available_moc_front_ends = get_moc_frontends(argv)
 
-    is_tf, is_caffe, is_mxnet, is_kaldi, is_onnx =\
-        deduce_framework_by_namespace(argv) if not moc_front_end else [False, False, False, False, False]
+    is_tf, is_caffe, is_mxnet, is_kaldi, is_onnx, is_paddle =\
```

Comment:
I thought the idea is that 'model optimizer' with new frontends should not know about actual frontends. I.e. MO shall not know about 'is_paddle' or something else - it is responsibility of FrontendManager.
@slyalin could you please comment on this? There was no 'is_paddle' in MO main code before

Reply:
This is needed to make the error message:
```
[ ERROR ]  It is not possible to use the --use_legacy_frontend flag when processing a PaddlePaddle model.
```
This was requested and discussed offline.

Reply:
Can we advice user to not use --legacy option for any framework not in the legacy list? And throw error "Cannot use --legacy option for {} framework".format(argv.framework)

Reply:
Discussed this subject offline. @nosovmik, please leave detailed comment "how to do better" to avoid `is_paddle` (minimize FW specific code)

Reply:
Added changes as discussed offline.